src/components/NewComment.jsx

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from "react";
+import { fetchUsers, postComment } from "../api";
+
+const NewComment = ({ id, setCommentsArr }) => {
+  const [formValues, setFormValues] = useState({
+    username: "",
+    commentBody: "",
+  });
+
+  const [usersArr, setUsersArr] = useState([]);
+  const [feedbackMessage, setFeedbackMessage] = useState({
+    posted: 0,
+    message: "",
+  });
+  const [formDisabler, setFormDisabler] = useState(false);
+
+  useEffect(() => {
+    fetchUsers()
+      .then(({ users }) => {
+        return users;
+      })
+      .then((users) => {
+        setUsersArr(users);
+        return users;
+      })
+      .then((users) => {
+        setFormValues((currValue) => {
+          return { ...currValue, username: users[0].username };
+        });
+      });
+  }, []);
+
+  const handleChange = (event) => {
+    if (
+      (event.target.id === "commentBody" && event.target.value.length >= 201) ||
+      event.target.value.startsWith(" ") ||
+      event.target.value.endsWith("  ")
+    ) {
+      return;
+    } else {
+      setFormValues((currValues) => {
+        return { ...currValues, [event.target.id]: event.target.value };
+      });
+    }
+  };
+
+  const submitForm = (event) => {
+    event.preventDefault();
+    setFormDisabler(true);
+
+    if (formValues.username !== "" && /^.{3,}$/i.test(formValues.commentBody)) {
+      postComment(id, formValues.username, formValues.commentBody)
+        .then(({ comment }) => {
+          setCommentsArr((currComments) => {
+            return [comment, ...currComments];
+          });
+          setFormValues((currValues) => {
+            return { ...currValues, commentBody: "" };
+          });
+        })
+        .then(() => {
+          setFormDisabler(false);
+          setFeedbackMessage({
+            posted: 1,
+            message: "Successfully posted comment.",
+          });
+        })
+        .catch((err) => {
+          console.log(err);
+          setCommentsArr((currComments) => {
+            return [
+              {
+                author: "ERROR",
```

Comment:
I see the connection now. A tad unusual to do it like so. We could simply have an error state and deal with it that way